Doc/library/datetime.rst

```diff
@@ -1219,10 +1219,10 @@ Instance methods:
    YYYY-MM-DDTHH:MM:SS.mmmmmm or, if :attr:`microsecond` is 0,
    YYYY-MM-DDTHH:MM:SS
 
-   If :meth:`utcoffset` does not return ``None``, a 6-character string is
-   appended, giving the UTC offset in (signed) hours and minutes:
-   YYYY-MM-DDTHH:MM:SS.mmmmmm+HH:MM or, if :attr:`microsecond` is 0
-   YYYY-MM-DDTHH:MM:SS+HH:MM
+   If :meth:`utcoffset` does not return ``None``, a string is
+   appended, giving the UTC offset:
+   YYYY-MM-DDTHH:MM:SS.mmmmmm+HH:MM[:SS[.uuuuuu]] or, if :attr:`microsecond`
+   is 0 YYYY-MM-DDTHH:MM:SS+HH:MM[:SS[.uuuuuu]].
```

Comment:
Would be more consistent to use _mmmmmm_ rather than _uuuuuu_.

Since you are introducing square brackets, you could eliminate the second form and just write [.mmmmmm] for both time and offset parts.

Reply:
Hm, we are wildly inconsistent in this document.

From the `fromisoformat` documentation:

>  Specifically, this function supports strings in the format(s)
>  ``YYYY-MM-DD[*HH[:MM[:SS[.mmm[mmm]]]][+HH:MM[:SS[.ffffff]]]]``,
>  where ``*`` can match any single character.

That uses both `fff` and `mmm` in the *same format string*.

[`moment.js`](http://momentjs.com/docs/#/displaying/format/) uses `SS` for fractional seconds, but I find that very confusing.

I think that `f` is most consistent with the `%f` notation in `str{f,p}time`, and avoids the question of whether it should be `[.mmm[.uuu]]`.